tests/mir-opt/switch_to_self.rs

```diff
@@ -1,3 +1,4 @@
+//@ unit-test: MatchBranchSimplification
```

Comment:
I think it's okay to change it for unit testing since the MIR output is the same.

Reply:
I'm surprised the test didn't already have this declaration, if it was meant to test a particular pass.

Reply:
Because the default compiler flag is `-O`.

Reply:
Yeah I know it doesn't *need* the declaration to test default-on opts. But when a test tests a specific opt, wouldn't it be good to add that declaration even if it is not needed?

Reply:
There are probably similar test cases in the repository. ;) Perhaps there was no support for unit testing in the past?

Reply:
For fixing the miscompilation, I hope to make as few changes as possible. This can go into a separate PR now.

Reply:
Oh yeah sure this was more of a general comment. Sorry for being unclear.